modules/csg/SCsub

```diff
@@ -6,6 +6,64 @@ Import("env_modules")
 env_csg = env_modules.Clone()
 
 # Godot source files
+
+if env_csg["disable_exceptions"]:
+    # Enable exceptions for only manifold.
+    if env_csg.msvc:
+        env_csg.Append(CPPDEFINES=[("_HAS_EXCEPTIONS", 1)], CCFLAGS=["/EHsc"])
+    else:
+        env_csg.Append(CXXFLAGS=["-fexceptions"])
+
+
+thirdparty_dir = "#thirdparty/manifold/"
+thirdparty_sources = [
+    thirdparty_dir + file
+    for file in [
+        "src/polygon/src/polygon.cpp",
+        "src/manifold/src/constructors.cpp",
+        "src/manifold/src/edge_op.cpp",
+        "src/manifold/src/face_op.cpp",
+        "src/manifold/src/impl.cpp",
+        "src/manifold/src/boolean_result.cpp",
+        "src/manifold/src/boolean3.cpp",
+        "src/manifold/src/manifold.cpp",
+        "src/manifold/src/properties.cpp",
+        "src/manifold/src/smoothing.cpp",
+        "src/manifold/src/sort.cpp",
+        "src/collider/src/collider.cpp",
+        "src/manifold/src/subdivision.cpp",
+        "src/manifold/src/csg_tree.cpp",
+        "src/cross_section/src/cross_section.cpp",
+        "src/third_party/quickhull/QuickHull.cpp",
```

Comment:
This is good practice for me, since I haven't seen a large project integrate our library this way before. I would like to make it easier if possible, and more modular. I take it you don't want to integrate us as a submodule? 

If you're not interested in 2D tools or convex hulls, it should be possible to drop those dependencies, but we could probably structure our library better for doing this. @pca006132 do you know any way to allow e.g. building Manifold without quickhull, so that the `Manifold::ConvexHull` methods are just stubs?

Reply:
Some kind of `#if` will work, change them to throwing exceptions when user don't want to compile quickhull.

Reply:
As a general rule, Godot Engine vendors its entire dependencies because it makes development much easier for contributors and for CI/CD. I am interested in the other libraries, but it would be best if I can find ways to shrink the binary size.

Although, if your quickhull is good, @lawnjelly has been in that area and can evaluate the quality.

Reply:
Would love to hear your evaluation - in fact we have a GSoC contributor who just started and is looking into improving our hulling - either using a different dependency or writing a new one. https://github.com/opencax/GSoC/issues/89

Reply:
To be honest my best experience is with https://github.com/SarahWeiii/CoACD for convex hulling.